huaweicloud/resource_huaweicloud_scm_certificate_v3_test.go

```diff
@@ -0,0 +1,228 @@
+)
+
+func TestAccScmCertificationV3_basic(t *testing.T) {
+	var certInfo certificates.CertificateEscrowInfo
+	resourceName := "huaweicloud_scm_certificate.certificate_1"
+
+	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	rUpdateName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheckScm(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckScmV3CertificateDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccScmCertificateV3_basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScmV3CertificateExists(resourceName, &certInfo),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "status", "UPLOAD"),
+				),
+			},
+			{
+				Config: testAccScmCertificateV3_basic(rUpdateName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScmV3CertificateExists(resourceName, &certInfo),
+					resource.TestCheckResourceAttr(resourceName, "name", rUpdateName),
+					resource.TestCheckResourceAttr(resourceName, "status", "UPLOAD"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccScmCertificationV3_push(t *testing.T) {
+	var certInfo certificates.CertificateEscrowInfo
+	resourceName := "huaweicloud_scm_certificate.certificate_2"
+
+	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	targetService := "Enhance_ELB"
+	defaultTargetProject := "ap-southeast-1"
+	newTargetProject := "la-south-2"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheckScm(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckScmV3CertificateDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccScmCertificateV3_push(rName, defaultTargetProject, targetService),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScmV3CertificateExists(resourceName, &certInfo),
+					testAccCheckScmV3CertificatePushExists(resourceName, targetService, defaultTargetProject),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "status", "UPLOAD"),
+				),
+			},
+			{
+				Config: testAccScmCertificateV3_push(rName, newTargetProject, targetService),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckScmV3CertificateExists(resourceName, &certInfo),
+					testAccCheckScmV3CertificatePushExists(resourceName, targetService, newTargetProject),
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "status", "UPLOAD"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccScmCertificationV3_batchPush(t *testing.T) {
+	var certInfo certificates.CertificateEscrowInfo
+	resourceName := "huaweicloud_scm_certificate.certificate_3"
+
+	rName := fmt.Sprintf("tf-acc-test-%s", acctest.RandString(5))
+	targetService := "Enhance_ELB"
+	defaultTargetProject := "ap-southeast-1"
```

Comment:
hard coding is not recommend, please use `HW_DEST_PROJECT` environment variable.

Reply:
In order to be consistent with other variables, use: `HW_CERTIFICATE_TARGET_SERVICE`, `HW_CERTIFICATE_TARGET_PROJECT` and `HW_CERTIFICATE_TARGET_PROJECTS` instead.